File: snippets/fsharp/System/Double/Parse/parse2.fs
// <Snippet3>
open System

[<EntryPoint>]
let main _ =
    let value = string Double.MinValue
    try
        printfn $"{Double.Parse value}"
    with :? OverflowException ->
        printfn $"{value} is outside the range of the Double type."

    let value = string Double.MaxValue
    try
        printfn $"{Double.Parse value}"
    with :? OverflowException ->
        printfn $"{value} is outside the range of the Double type."

    // Format without the default precision.
    let value = Double.MinValue.ToString "G17"
    try
        printfn $"{Double.Parse value}"
    with :? OverflowException ->
        printfn $"{value} is outside the range of the Double type."

    0
// The example displays the following output:
//    -1.79769313486232E+308 is outside the range of the Double type.
//    1.79769313486232E+308 is outside the range of the Double type.
//    -1.79769313486232E+308
// </Snippet3>

File: snippets/csharp/System/Double/Parse/parse2.cs
using System;

public class Example
{
    public static void Main()
    {
        // <Snippet3>
        string value;

        value = double.MinValue.ToString();
        try
        {
            Console.WriteLine(double.Parse(value));
        }
        catch (OverflowException)
        {
            Console.WriteLine($"{value} is outside the range of the Double type.");
        }

        value = double.MaxValue.ToString();
        try
        {
            Console.WriteLine(double.Parse(value));
        }
        catch (OverflowException)
        {
            Console.WriteLine($"{value} is outside the range of the Double type.");
        }

        // Format without the default precision.
        value = double.MinValue.ToString("G17");
        try
        {
            Console.WriteLine(double.Parse(value));
        }
        catch (OverflowException)
        {
            Console.WriteLine($"{value} is outside the range of the Double type.");
        }
        // The example displays the following output:
        //    -1.79769313486232E+308 is outside the range of the Double type.
        //    1.79769313486232E+308 is outside the range of the Double type.
        //    -1.79769313486232E+308
        // </Snippet3>
    }
}
